### PR TITLE
skip `allow_superuser` in Windows OS

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

[rn:skip]

## What does this PR do?

As Process.euid() is always zero in Windows,
this commit excluded the checking of running as root in Windows.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- relates: https://github.com/elastic/logstash/pull/14089 , https://github.com/elastic/logstash/pull/16558

### CODE REVIEW

**Comment by kaisecheng** [comment]
windows buildkite is [green](https://buildkite.com/elastic/logstash-windows-jdk-matrix-pipeline/builds/264#_)

File: config/logstash.yml [hunk 1]
```diff
@@ -320,7 +320,7 @@
 #
 # ------------ Other Settings --------------
 #
-# Allow or block running Logstash as superuser (default: true)
+# Allow or block running Logstash as superuser (default: true). Windows are excluded from the checking
 # allow_superuser: false
```
Comment (mashhurs) [inline config/logstash.yml:324]: Did we agree that we are not going to support this feature on Windows?

Comment (donoghuc) [reply config/logstash.yml:324]: If there is a desire to generally determine if a process is running as an admin or with elevated status on windows there is a pattern in Puppet for doing so https://github.com/puppetlabs/puppet/blob/e758d5c969403631810fa6385057ef0eaf03974f/lib/puppet/util/windows/user.rb#L11 though it is fairly complex and carries a dependency on ffi which requires native extensions.

Comment (kaisecheng) [reply config/logstash.yml:324]: @mashhurs My goal here is to unblock the exhaustive test, not to bring the feature to windows
I believe it is another PR to fix the original handling

**Review by mashhurs — APPROVED** [review]
LGTM!
Feature bug will be fixed/handlied with https://github.com/elastic/logstash/issues/16632

**Pull request merged**